src/linter/ui5Types/BestPractices.ts

```diff
@@ -0,0 +1,345 @@
+
+	const compareValues = (aProp: null | undefined | boolean,
+		bProp: null | undefined | boolean): null | undefined | boolean => {
+		let result = null;
+		if (aProp === undefined || bProp === undefined) {
+			result = undefined;
+		}
+		if (typeof aProp === "boolean" || typeof bProp === "boolean") {
+			// eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+			result = aProp || bProp;
+		}
+
+		return result;
+	};
+
+	return {
+		hasManifestDefinition: a.hasManifestDefinition || b.hasManifestDefinition,
+		routingAsyncFlag: compareValues(a.routingAsyncFlag, b.routingAsyncFlag),
+		rootViewAsyncFlag: compareValues(a.rootViewAsyncFlag, b.rootViewAsyncFlag),
+		hasAsyncInterface: compareValues(a.hasAsyncInterface, b.hasAsyncInterface),
+	};
+}
+
+function findAsyncInterface({classDefinition, manifestContent, checker, uiComponentImportVar}: {
+	classDefinition: ts.ClassDeclaration;
+	manifestContent: string | undefined;
+	checker: ts.TypeChecker;
+	uiComponentImportVar: string;
+}): AsyncInterfaceFindType | undefined {
+	if (!ts.isClassDeclaration(classDefinition)) {
+		return;
+	}
+
+	const returnTypeTemplate = {
+		hasAsyncInterface: null,
+		routingAsyncFlag: null,
+		rootViewAsyncFlag: null,
+		hasManifestDefinition: false,
+	} as AsyncInterfaceFindType;
+
+	// Checks the interfaces and manifest
+	const curClassAnalysis = classDefinition.members.reduce((acc, member) => {
+		const checkResult = doPropsCheck(member as ts.PropertyDeclaration, manifestContent);
+		return mergeResults(acc, checkResult);
+	}, {...returnTypeTemplate});
+
+	const heritageAnalysis =
+		classDefinition?.heritageClauses?.flatMap((parentClasses: ts.HeritageClause) => {
+			return parentClasses.types.flatMap((parentClass) => {
+				const parentClassType = checker.getTypeAtLocation(parentClass);
+
+				return parentClassType.symbol?.declarations?.flatMap((declaration) => {
+					let result = {...returnTypeTemplate} as AsyncInterfaceFindType;
+					// Continue down the heritage chain to search for
+					// the async interface or manifest flags
+					if (ts.isClassDeclaration(declaration) &&
+						(!uiComponentImportVar || declaration?.name?.getText() !== uiComponentImportVar)) {
+						result = findAsyncInterface({
+							classDefinition: declaration,
+							manifestContent,
+							checker,
+							uiComponentImportVar,
+						}) ?? result;
+					} else if (ts.isInterfaceDeclaration(declaration)) {
+						result.hasAsyncInterface = doAsyncInterfaceChecks(parentClass) ?? result.hasAsyncInterface;
+					}
+
+					return result;
+				});
+			});
+		}) ?? [];
+
+	return [...heritageAnalysis, curClassAnalysis].reduce((acc, curAnalysis) => {
+		return mergeResults(acc ?? {...returnTypeTemplate}, curAnalysis ?? {...returnTypeTemplate});
+	});
+}
+
+function doAsyncInterfaceChecks(importDeclaration: ts.Node): boolean | undefined {
+	while (!importDeclaration || importDeclaration.kind !== ts.SyntaxKind.SourceFile) {
```

Comment:
Wouldn't `!importDeclaration` lead to an exception once a node has an undefined parent (unable to access parent on it)? Am I missing something? 😅 

Did you mean something like:
```suggestion
	while (importDeclaration && importDeclaration.kind !== ts.SyntaxKind.SourceFile) {
```

Reply:
Done